internal/cliwrapper/cliwrapper.go

```diff
@@ -7,30 +7,67 @@ import (
 	"go.arcalot.io/log/v2"
 	"go.flow.arcalot.io/pythondeployer/internal/models"
 	"io"
-	"os"
 	"os/exec"
+	"path/filepath"
 	"regexp"
 	"strings"
+	"sync"
 )
 
 type cliWrapper struct {
 	pythonFullPath string
-	workDir        string
+	pluginDir      string
+	connectorDir   string
 	deployCommand  *exec.Cmd
 	logger         log.Logger
-	stdErrBuff     bytes.Buffer
+	stdErrBuff     bufferThreadSafe
+}
+
+type bufferThreadSafe struct {
+	b    bytes.Buffer
+	lock sync.Mutex
```

Comment:
I'm not 100% confident of the Go semantics here. You're passing in a `sync.Mutex{}`; does that end up copying the mutex object to the field, or is the compiler smart enough to create `lock` directly? And if it's a copy, is copying a Go `Mutex` object legal?

This question becomes moot if you add a constructor function for `bufferThreadSafe`, which additionally encapsulates the mutex (which I don't think you really want to expose outside the "object" anyway). And ... although I'm not entirely sure it would be a big benefit, that would allow transparently considering changes like using an `RWLock` so that `Len` and `String` could run in parallel.

Reply:
> You're passing in a `sync.Mutex{}`; does that end up copying the mutex object to the field[...]?

Yes, I believe it copies the contents of the struct; however, this is fine, because the value being copied is a "zero value" (i.e., a mutex which has never been locked).

> This question becomes moot if you add a constructor function for bufferThreadSafe, 

With the substructures _contained_ in `bufferThreadSafe`, it shouldn't need a constructor function -- just referencing `bufferThreadSafe{}` should suffice.

Reply:
> > You're passing in a `sync.Mutex{}`; does that end up copying the mutex object to the field[...]?
> 
> Yes, I believe it copies the contents of the struct; however, this is fine, because the value being copied is a "zero value" (i.e., a mutex which has never been locked).

Sure; but initializing one `struct` just to copy to another is, at best, wasteful. And, as I said, the buffer could easily be refactored to use r/w locks without any external impact if this were changed (another good indication that encapsulation with a constructor would be more appropriate).